sys/kern/sysv_shm.c

```diff
@@ -526,6 +526,12 @@ kern_shmat_locked(struct thread *td, int shmid,
 			cow |= MAP_REMAP;
 		find_space = VMFS_NO_SPACE;
 	} else {
+		/*
```

Comment:
Using this hint for `attach_va` for CheriABI is the only non-trivial change I believe.  I think we want this to avoid using the address of userspace_root_cap (probably 0) as the hint instead.

Reply:
/me notes hoisting this up here actually marginally reduces the upstream diff